ion/symboltoken.go

```diff
@@ -92,6 +92,19 @@ func (st *SymbolToken) Equal(o *SymbolToken) bool {
 	return false
 }
 
+// ResolveToString returns a SymbolToken's text value or "$<sid>"
+func (st *SymbolToken) ResolveToString() string {
```

Comment:
The motivation behind this function is for getting some ion-hash tests passing in https://github.com/amzn/ion-hash-go/pull/62.

```go
val, err := reader.SymbolValue()
assert.NoError(t, err, "Something went wrong when reading Symbol value")

assert.NoError(t, writer.WriteSymbol(val.ResolveToString()), "Something went wrong when writing Symbol value")
```

Note that `writer.go` currently has `WriteSymbol(val string)` not `WriteSymbol(val SymbolToken)`.

Reply:
> The motivation behind this function is for getting some ion-hash tests passing 

My gut reaction is that this method is not serving a common use case outside of our own libraries and so shouldn't be part of the public API we commit to in v1.0. Could/should this just be a private helper function in the `ion-hash-go` test suite?

> Note that writer.go currently has WriteSymbol(val string) not WriteSymbol(val SymbolToken).

I thought the takeaway from last week's meeting was that we'd be using `SymbolToken` in all of the `Reader`/`Writer` APIs that involve symbols so we'd have a consistent (if a bit tedious) error reporting story.

Reply:
I took your suggestion and moved the ResolveToString() logic to ion-hash-go. After making that change the rest of this PR is basically pointless since the annotations API is being redone by Byron anyway.

I will close this PR and let Byron's PR handle the annotations API changes.